test/calculator_test.js

```diff
@@ -0,0 +1,44 @@
+const assert = require('assert');
+const chai = require('chai');
+const calc = require('../calculator')
+
+describe('calculator', function () {
+  describe('#calculate()', function () {
+    it('adds one number', function () {
+      const calculator = calc.create_calculator('+');
+
+      calculator.add_number(16);
+
+      assert.equal(16, calculator.calculate());
+    });
+
+    it('adds two numbers', function () {
+      const calculator = calc.create_calculator('+');
+
+      calculator.add_number(5);
+      calculator.add_number(4);
+
+      assert.equal(9, calculator.calculate());
+    });
+
+    it('adds many numbers', function () {
```

Comment:
Notably a decimal would cause the test to fail since the source code uses 'parseInt' instead of 'parseFloat'. Of course that's not a fault of the test; the source code is at fault for being too restrictive on allowing only integer inputs (which is especially weird given that at least in the case of division, the program is happy outputting floats).